drivers/serial/uart_stm32.c

```diff
@@ -2123,9 +2057,17 @@ static int uart_stm32_init(const struct device *dev)
 		return err;
 	}
 
-#if defined(CONFIG_PM) || \
-	defined(CONFIG_UART_INTERRUPT_DRIVEN) || \
-	defined(CONFIG_UART_ASYNC_API)
+/*Init deassert timer and pin*/
+#ifdef CONFIG_SERIAL_SUPPORT_RS485
+	if (config->de_enable) {
+		gpio_pin_configure_dt(&config->de_pin, GPIO_ACTIVE_LOW | GPIO_OUTPUT_ACTIVE);
+		gpio_pin_set(config->de_pin.port, config->de_pin.pin, config->de_invert);
+		k_timer_init(&data->rs485_timer, rs485_de_time_expire_callback, NULL);
+		k_timer_user_data_set(&data->rs485_timer, (void *)config);
+	}
```

Comment:
I wonder why using a GPIO DE bit instead of using the built in DE pin functionality driver by HW (and already implemented, by the way).
At the minimum the PR should explain why introducing this alternate software control method, why one would use it and make it optional (if ever there is a really good reason to have it in the driver)

Reply:
Thank you very much for your response! If I understand correctly, not all STM32 controllers have hardware support for RS485 (e.g., F0, F1, F4 don't have it), and therefore it is not possible to control the DE pin via hardware on these models. I agree that the API should support hardware control (if the controller supports it), and this needs to be implemented.

My solution proposes an alternative method for controlling the pin for RS485 and a unified API, which, as I understand it, is currently missing in Zephyr. RS485 can be used not only for Modbus, after all. As far as I can tell, people often have the need to use RS485 for other purposes, for example (https://github.com/zephyrproject-rtos/zephyr/issues/32733).

I would appreciate any advice or guidance to successfully integrate this solution.

Reply:
> I agree that the API should support hardware control (if the controller supports it), and this needs to be implemented.

My point is that hw control is already implemented in this driver. See https://github.com/zephyrproject-rtos/zephyr/blob/5f418f506c717ed39686ab0b11a3e6e4bf4fb756/drivers/serial/uart_stm32.c#L2028-L2043

Hence your implementation should take it into account and ensure it could be used by default over SW control if present.